connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -141,17 +141,35 @@ public void initialize(TaskConfig taskConfig) {
     public void stop() {
         // Offset commit is handled upon exit in work thread
         super.stop();
+        try {
+            task.stop();
+        } catch (Exception e) {
+            log.warn("Could not stop task", e);
+        }
```

Comment:
Should this be called before or after the `consumer.wakeup()`? Is there any chance the consumer returns more information and the worker's thread calls `task.put` after this thread calls `task.stop`? I don't think we have documented the semantics of the interaction between stop and a subsequent put, so should perhaps be more careful here.

Reply:
The next method `close()` calls `task.stop()` followed by `consumer.close()`.   For consistency I think I will try`task.stop()` followed by `consumer.wakeup()` unless testing reveals that it is problematic.